### DISCUSSION TITLE
Summer mode exhaust fan

### DISCUSSION THREAD

**Discussion by SgtFoose** [adorobis/hacomfoairmqtt#62] [General]
Hi,

Im reading through your code to find the hex Command to change the inlet and outlet fan.

I’d like to add a button to set the summer mode in Home Assistant. I’m not using MQTT but direct command line commands.

the protocol in German does not clearly state the command?

have you found it?

I’m using these for example to change the fan speeds:

shell_command:
  run_nc_command_fan_1: echo -e -n '\x07\xF0\x00\x99\x01\x02\x49\x07\x0F' | nc 192.168.178.22 8899
  run_nc_command_fan_2: echo -e -n '\x07\xF0\x00\x99\x01\x03\x4A\x07\x0F' | nc 192.168.178.22 8899
  run_nc_command_fan_3: echo -e -n '\x07\xF0\x00\x99\x01\x04\x4B\x07\x0F' | nc 192.168.178.22 8899

your help is much appreciated!

**Comment by adorobis** [comment]
Generally there is no such command to control summer/winter mode switch. The unit decides on it itself based on outside temperature.
What you have on the unit is the out only fans (in reality, the intake fan still runs but on the lowest setting, at least in the CA350 unit). And the way it is controlled is to change the settings of fan levels - when out only (or in only) is selected, the setting of fan speed for intake fan is changed to the lowest possible (15% in my unit - btw, this is described in the unit installation manual). And when switching back to both in and out fans, the setting of the previously "disabled" fans are restored to pre-configured values.
You can find the logic in the `set_fan_levels` [function ](https://github.com/adorobis/hacomfoairmqtt/blob/master/src/ca350.py#L499)and the respective hex codes (well, not directly, but you'll find there how they are calculated).

**Reply by SgtFoose** [reply]
Thanks for this info! Is there also a direct command to disable the inlet fan? If yes, to turn it back on, I assume it requires a value?

**Reply by adorobis** [reply]
No, that's what I was trying to explain. There is no such command, you need to send command to re-configure fan speeds of each level so that it stays on the lowest level. And when you want to "enable" it again you need to re-configure it to the desired speeds for each level.

**Reply by SgtFoose** [reply]
Ok I was running this code on a Pi2 with crashes so bought a pi4b and fixed to code to make it run. First it had many errors but now code is improved and working again. Also found a better way to parse feedback. I have a Storkair WHR930 instead of a CA350 but code is compatible. My unit seems to respond well on all available commands now but noticed the unit itself does not always update to the correct feedback on the display. Stuff like changing the comfort temperature via the gauge in HA does actually work and run, but you need to sometimes go in and out of the menu on the unit so see the actual status. This is a minor inconvenience so no problem. What I'm trying to achieve is to force the unit to disable the inlet and make it extract air only. In winter times neighbors have smelly fireplaces and also fireworks stink, that's why. Changing the dropdown to outlet seems to work, it's just the units display that doesn't update its status. I need to play little more to figure it out I guess. My unit does not have the anti freeze pre-heating element in it. Is that what the "EWT" does?

**Reply by adorobis** [reply]
EWT is the geothermal heat exchanger. Usually in a form of a long intake pipe buried in the garden, also there are some other forms of it. If you don't have it the unit will just ignore whatever parameters you'll set there.

> bought a pi4b and fixed to code to make it run

What fixes have you made? Is it to the python code or something else?

> the unit itself does not always update to the correct feedback on the display

Which display do you have?
